envoy/stream_info/stream_info.h

```diff
@@ -732,6 +732,10 @@ class StreamInfo {
    */
   virtual void setDownstreamBytesMeter(const BytesMeterSharedPtr& downstream_bytes_meter) PURE;
 
+  virtual void setPathWithoutQueryAndFragment(const Http::RequestHeaderMap& headers) PURE;
```

Comment:
Please add comments for the interface.

A different question, why not set the entire path in the stream info, and allow fetching the path without the query and fragment parts?

Reply:
It's actually the result of `removeQueryAndFragment` will affect the performance. I think set the entire path in the stream info is unnecessary?